### DISCUSSION TITLE
Center string  with spaces

### DISCUSSION THREAD

**Discussion by rradomir** [Bodmer/TFT_eSPI#3001] [Q&A - General]
Hi, I want to use drawString with char * table. It's only 4 elements long. Sometimes all 4 digits are used, sometimes only 2 with spaces at end. Still, setTextDatum(TC_DATUM); and drawString will center all string with spaces, not only 2 digits. How to properly do it?

**Comment by Bodmer** [comment]
Post an example that demonstrates the problem.

**Comment by larrybud2004** [comment]
I guess my question is why do you think it should automatically trim the string for you? How does it know you really want that string trimmed? I'd say trim starting or trailing spaces yourself if that's what you really want to do.
